diesel/src/pg/types/numeric.rs

```diff
@@ -0,0 +1,118 @@
+#[cfg(feature="bigdecimal")]
+mod bigdecimal {
+    extern crate num_traits;
+    extern crate num_bigint;
+    extern crate num_integer;
+    extern crate bigdecimal;
+
+    use std::error::Error;
+    use std::io::prelude::*;
+
+    use pg::Pg;
+
+    use self::num_traits::{Signed, Zero, ToPrimitive};
+    use self::num_bigint::{Sign, BigInt, BigUint, ToBigInt};
+    use self::num_integer::Integer;
+    use self::bigdecimal::BigDecimal;
+
+    use pg::data_types::PgNumeric;
+    use types::{self, FromSql, ToSql, IsNull};
+
+    impl ToSql<types::Numeric, Pg> for BigDecimal {
+        fn to_sql<W: Write>(&self, out: &mut W) -> Result<IsNull, Box<Error + Send + Sync>> {
```

Comment:
Can you add a comment here giving an overview of what happens here and why encoding the BigDecimal is a bit complicated?